calico-enterprise/operations/monitor/metrics/recommended-metrics.mdx

```diff
@@ -0,0 +1,574 @@
+| Example value                    | <code>process_resident_memory_bytes{**endpoint**="metrics-port", **instance**="10.0.1.20:9093", **job**="typha-metrics-svc", **namespace**="calico-system", **pod**="calico-typha-6c6cc9fcf7-csbdl", **service**="typha-metrics-svc"} 80515072</code> |
+| Explanation                      | Amount of memory used by Typha.                              |
+| Threshold value recommendation   | It is recommended to achieve a baseline first, then monitor for any unexpected increases from this baseline. A rule of thumb is to investigate if maintained memory usage is above 90% of what is available from the underlying node. The metric can also be used for memory leaks. In this case, the metric would show Typhas' memory consumption rising over time, even though the cluster is in a stable state. |
+| Threshold breach symptoms        | Unexpected maintained memory usage could cause Typha to fall behind in updating its clients (for example, Felix) and could cause delays to policy updates. |
+| Threshold breach recommendations | Check memory usage on Kubernetes nodes. Increase resources if needed, and rollout restart Typha(s) if needed. |
+| Priority level                   | Recommended.                                                 |
+
+## Typha cluster mesh metrics
+
+The following metrics are applicable only if you have implemented [Cluster mesh](/multicluster/federation/overview.mdx).
+
+Note that this metric requires a count syntax because you will have a copy of the metric per RemoteClusterConfiguration. As shown in the table, the value `2 = In Sync` reflects good connections.
+
+```
+remote_cluster_connection_status{cluster="foo"} = 2
+remote_cluster_connection_status{cluster="bar"} = 2
+remote_cluster_connection_status{cluster="baz"} = 1
+```
+
+### Remote cluster connections (in-sync)
+
+| Remote cluster connections (in-sync) |                                                              |
+| ------------------------------------ | ------------------------------------------------------------ |
+| Metric                               | <code>count by (instance) (remote_cluster_connection_status == 2)</code> |
+| Explanation                          | This represents the number of remote cluster connections that are connected and in sync. Each remote cluster will report a *connection_status* value from the following list:<br />- 0 = Not Connected<br />- 1 = Connecting<br />- **2 = In Sync**<br />- 3 = Resync in Process<br />- 4 = Config Change Restart Required<br /><br />We suggest the **count** syntax because there will be one copy of *remote_cluster_connection_status* per cluster: - <code>remote_cluster_connection_status[cluster="foo"] = 2</code><br /><code>remote_cluster_connection_status[cluster="bar"] = 2</code><br /><code>remote_cluster_connection_status[cluster="baz"] = </code>'<br /><br />Counting the number of metrics with value **2** returns the number of **In Sync** clusters. |
```

Comment:
```suggestion
| Explanation                          | This represents the number of remote cluster connections that are connected and in sync. Each remote cluster will report a *connection_status* value from the following list:<br />- 0 = Not Connected<br />- 1 = Connecting<br />- **2 = In Sync**<br />- 3 = Resync in Process<br />- 4 = Config Change Restart Required<br /><br />We suggest the **count** syntax because there will be one copy of *remote_cluster_connection_status* per cluster: - <code>remote_cluster_connection_status[cluster="foo"] = 2</code><br /><code>remote_cluster_connection_status[cluster="bar"] = 2</code><br /><code>remote_cluster_connection_status[cluster="baz"] = 2</code><br /><br />Counting the number of metrics with value **2** returns the number of **In Sync** clusters. |
```

Reply:
How about using the presentation you used for Datastore cache size. I think that works well, using the pattern of "Here are the raw metrics, and here's how we recommend that you use them...":

> Individual metrics values:
>
> remote_cluster_connection_status[cluster="cluster-A"] = 2
> remote_cluster_connection_status[cluster="cluster-B"] = 2
>
> Count of number of in-sync clusters:
>
> count by (instance) (remote_cluster_connection_status == 2)

could even combine the in-sync and out-of-sync blocks.